### DISCUSSION TITLE
Please, fix documentation. Need more info about authentication

### DISCUSSION THREAD

**Discussion by olegKusov** [vercel/next.js#23786] [Help]
I'm making blog. created backend with nodejs and mysql.  know I want to integrate my backend auth endpoints with nextjs and can't find solution. official documentation give us vague explanation. There is no example with custom backend server. Just examples with third party libs like authjs. I don't need authjs or smth else, I need just simple example with simple backend. And there is no solution or explanation about it in documentation.

1. How to store access and refresh key in NextJs?
2. How to handle  protected routes?
3. Auth flow with nextjs + custom backend auth server? How to refresh token when user close tab and open new tab?

There is a lot of questions.

**Comment by barelyhuman** [comment]
You can still use HOC's and or Hooks to help you with this use case. 

Have a HOC that manages the authentication state, or checks the localStorage/cookies/request server for auth check and based on that either render the component or redirect the user to the login page. 

the same can be done with a custom hooks which manages the authenticated state, it being `false` to start with and same as above checking your auth state and redirecting accordingly. 

Let's say you don't use httpOnly cookie 

```js
import useQuery from 'hooks/use-query';
import Router, { useRouter } from 'next/router';
import React, { useEffect } from 'react';
import Cookies from 'universal-cookie';

export default function withAuth(WrappedComponent) {
  return (props) => {
    const router = useQuery();

    useEffect(() => {
      if (router) {
        const movingFrom = router.asPath;
        const token = new Cookies().get('auth');
        if (!token) {
          Router.push(`/login?redirect=${movingFrom}`);
        }
      }
    }, [router]);

    return <WrappedComponent {...props} />;
  };
}
```

This is generally how you do it in all forms of react apps and I'm assuming next.js didn't want to change that and hence left it on how the developer would like to implement it.
